tests/tests/tier0/bluechi-kill-unit/test_bluechi_kill_unit.py

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+from typing import Dict
+
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.service import Option, Section, SimpleRemainingService
+from bluechi_test.test import BluechiTest
+
+LOGGER = logging.getLogger(__name__)
+NODE_FOO = "node-foo"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+
+    sleep_service = SimpleRemainingService(name="sleep.service")
+    sleep_service.set_option(Section.Service, Option.ExecStart, "/bin/sleep 1000")
+    node_foo.install_systemd_service(sleep_service)
+
+    ctrl.bluechictl.start_unit(NODE_FOO, sleep_service.name)
```

Comment:
I think that 3 integration test files should be added.
For each of the kill options (all, main, control).
If the main option is given you want to check that only the main process was killed.
In addition, I would add another integration test for a signal check.
And maybe you should add for this integration test more than one service and check that only one service changed the state.

Reply:
I extended the current integration test by using all different options for `--kill-whom`. I don't want to add to many integration tests since these are rather costly for the setup - and these whom options are systemd logic, so testing these doesn't provide much benefit.